build.gradle

```diff
@@ -384,6 +386,28 @@ subprojects {
         }
     }
 
+    dependencyCheck {
+        scanConfigurations = [
+            'compile',
+            'implementation',
+            'compileOnly',
+            'compileClasspath',
+            'runtime',
+            'runtimeOnly',
+            'runtimeClasspath',
+            'default',
+        ]
+        analyzers {
+            assemblyEnabled = false
+        }
+        suppressionFile = file("${rootDir}/OWASP-Suppression.xml")
+
+        // Configures the dependencyCheck plugin to fail the build if any vulnerability with a CVSS score over the configured value is found
+        failBuildOnCVSS = 5
+    }
+
+    check.dependsOn dependencyCheckAnalyze
```

Comment:
Wiring this task into the standard build adds a fair amount of time to the build. Of course, you can always exclude the task so it should be fine.

Reply:
Here are the timings on my machine with and without the `dependencyCheckAnalyze` task.

Without, the build took 5m 46s:
```
BUILD SUCCESSFUL in 5m 46s
985 actionable tasks: 841 executed, 144 up-to-date

~/dev/repose REP-7150* 6m 2s
❯ gradlew clean && gradlew build -x dependencyCheckAnalyze -x test -x linkchecker
```

With, the build took 16m 16s:
```
BUILD SUCCESSFUL in 16m 16s
1092 actionable tasks: 948 executed, 144 up-to-date

~/dev/repose REP-7150* 16m 26s
❯ gradlew clean && gradlew build -x test -x linkchecker
```

Reply:
With those numbers being what they are, I am going to remove the wiring of this check into the main build task chain. Instead, this task can be invoked explicitly.

Reply:
That makes sense. We can easily add an additional check on the nightly that invokes just the `dependencyCheckAnalyze` task.

Reply:
Yeah, I'd say add it into the nightly as well as giving it it's pr job.